assets/js/sections/contesting.js

```diff
@@ -208,20 +214,35 @@ $('#start_date').change(function () {
 
 // On Key up check and suggest callsigns
 $("#callsign").keyup(function () {
-	var call = $(this).val();
+	var call = $(this).val().toUpperCase();
 	if (call.length >= 3) {
 
-		$.ajax({
-			url: 'lookup/scp',
-			method: 'POST',
-			data: {
-				callsign: $(this).val().toUpperCase()
-			},
-			success: function (result) {
-				$('.callsign-suggestions').text(result);
-				highlight(call.toUpperCase());
-			}
-		});
+        if (scp_data.status < 0 || scp_data.request != call.substr(0, scp_data.request.length)) {
+          scp_data.status = 0;
+          scp_data.request = call;
+          scp_data.data = [];
+          $.ajax({
+              url: 'lookup/scp',
+              method: 'POST',
+              data: {
+                  callsign: call
+              },
+              success: function (result) {
+                  scp_data.status = 1;
+                  scp_data.data = result.split(" ");
+
+                  var call = $("#callsign").val().toUpperCase(); // Might have changed while running the query...
```

Comment:
I think shouldnt need to redeclare `call` here if you use `let` above`

Reply:
As stated above and as in the comment, the value could have changed by the time, the success-callback is being executed.

Reply:
youre right, but then if it has changed, shouldnt we just set the cache and not doing anything else?  I feel like if we move to the suggestions at that point there will be a mismatch from the call entered in the input and the results?